toolkit/scripts/build_cargo_cache.sh

```diff
@@ -2,60 +2,150 @@
+            echo "Error: Argument for $1 is missing" >&2
+            exit 1
+        fi
+        ;;
+        --vendorVersion)
+        if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+            VENDOR_VERSION=$2
+            shift 2
+        else
+            echo "Error: Argument for $1 is missing" >&2
+            exit 1
+        fi
+        ;;
+        -*|--*=) # unsupported flags
+        echo "Error: Unsupported flag $1" >&2
+        exit 1
+        ;;
+        *) # preserve positional arguments
+        PARAMS="$PARAMS $1"
+        shift
+        ;;
+  esac
+done
+
+echo "--srcTarball      -> $SRC_TARBALL"
+echo "--outFolder       -> $OUT_FOLDER"
+echo "--pkgVersion      -> $PKG_VERSION"
+echo "--vendorVersion   -> $VENDOR_VERSION"
+
+if [ -z "$PKG_VERSION" ]; then
+    echo "--pkgVersion parameter cannot be empty"
+    exit 1
+fi
+
+if [ -z "$VENDOR_VERSION" ]; then
+    echo "--vendorVersion parameter cannot be empty"
+    exit 1
 fi
 
-cache_tarball_name="$cache_name-cargo.tar.gz"
+echo "-- create temp folder"
+TEMP_DIR=$(mktemp -d)
+function cleanup {
+    echo "+++ cleanup -> remove $TEMP_DIR"
+    rm -rf $TEMP_DIR
+}
+trap cleanup EXIT
+
+pushd $TEMP_DIR > /dev/null
+
+TARBALL_NAME=$(basename "$SRC_TARBALL")
 
-if [[ $# -ge 2 ]]
+NAME_VER=${TARBALL_NAME%.*}
+if [[ "$NAME_VER" =~ \.tar$ ]]
 then
-    directory_name=$2
-else
-    directory_name=$cache_name
+    NAME_VER=${NAME_VER%.*}
 fi
 
-if [[ -f "$tarball_name" ]]
+VENDOR_TARBALL="$NAME_VER-cargovendor-v$VENDOR_VERSION.tar.gz"
+
+if [[ -f "$TARBALL_NAME" ]]
 then
-    cp "$tarball_name" "$temp_dir"
+    cp "$SRC_TARBALL" "$TEMP_DIR"
 else
-    echo "Tarball '$tarball_name' doesn't exist. Will attempt to download from blobstorage."
-    if ! wget -q "https://azurelinuxsrcstorage.blob.core.windows.net/sources/core/$tarball_name" -O "$temp_dir/$tarball_name"
+    echo "Tarball '$TARBALL_NAME' doesn't exist. Will attempt to download from blobstorage."
+    if ! wget -q "https://azurelinuxsrcstorage.blob.core.windows.net/sources/core/$TARBALL_NAME" -O "$TEMP_DIR/$TARBALL_NAME"
     then
         echo "ERROR: failed to download the source tarball."
         exit 1
     fi
     echo "Download successful."
 fi
 
-pushd "$temp_dir" &> /dev/null
-    echo "Extracting $tarball_name."
-    tar -xf "$tarball_name"
-    
-    pushd "$directory_name" &> /dev/null
-        echo "Fetching dependencies to a temporary cache."
-        CARGO_HOME=$(pwd)/.cargo cargo fetch
+echo "Unpacking source tarball..."
+tar -xf $SRC_TARBALL
+
+echo "Vendor cargo ..."
+DIRECTORY_NAME=($(ls -d */))
+
+# assume there is only one directory in the tarball
+DIRECTORY_NAME=${DIRECTORY_NAME[0]%//}
+
+pushd "$DIRECTORY_NAME" &> /dev/null
+echo "Fetching dependencies to a temporary cache in $DIRECTORY_NAME."
+
+#TODO: installing rust as auto-patcher does not have it installed by default. Possibly remove if auto-patcher will be changed to have rust included.
+sudo dnf install -y rust || sudo apt install -y rustc
```

Comment:
Hmm, I am not entirely sure if it make sense to have this in the script itself or move it out into pipeline code? I think generally we keep it in pipeline code, but might make it easier for anyone running this manually too